### PR TITLE
Fix stringified JSON response (#22450)

### LINKED ISSUE #22450: [Bug]: JSON response gets stringified by server
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

[Slack Conversation](https://theappsmith.slack.com/archives/C0341RERY4R/p1681123872154409)

Error in JSON format returned by users' API server gets stringified by Appsmith server.

### Steps To Reproduce

1. Create a REST API e.g. `Api1`
2. Enter `https://mock.codes/500` in the URI
3. Drag and drop a text widget and use `{{Api1.data.description}}` in the `Text` field

You will notice that the evaluated value doesn't show anything because the response is coming as stringified. 

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud / Self Hosted

### PR DESCRIPTION
**Pull request by subrata71**
## Description
TL;DR

This commit fixes unexpected stringified JSON in the following plugins
- REST API Plugin
- GraphQL plugin

The detailed analysis can be found in the comment section of the issue itself (#22450)

Slack Conversations
- [1](https://theappsmith.slack.com/archives/C0341RERY4R/p1681123872154409)
- [2](https://theappsmith.slack.com/archives/CGBPVEJ5C/p1681726446635249)

> Add a TL;DR when description is extra long (helps content team)

Fixes #22450 

Media

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Review by nidhi-nair — APPROVED** [review]

**Comment by subrata71** [comment]
@Rhitottam Can you please take a look at the perf-tests failure?
